core/templates/dev/head/domain/statistics/LearnerAnswerInfoObjectFactory.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of frontend learner answer info
+ * domain objects.
+ */
+
+var oppia = require('AppInit.ts').module;
+
+oppia.factory('LearnerAnswerInfoObjectFactory', [
+  function() {
+    var LearnerAnswerInfo = function(learnerAnswerInfoId, answer,
+        answerDetails, createdOn) {
+      this._id = learnerAnswerInfoId;
+      this._answer = answer;
+      this._answerDetails = answerDetails;
+      this._createdOn = createdOn;
+    };
+
+    LearnerAnswerInfo.prototype.getId = function() {
+      return this._id;
+    };
+
+    LearnerAnswerInfo.prototype.getAnswer = function() {
+      return this._answer;
+    };
+
+    LearnerAnswerInfo.prototype.getAnswerDetails = function() {
+      return this._answerDetails;
+    };
+
+    LearnerAnswerInfo.prototype.getCreatedOn = function() {
+      return this._createdOn;
+    };
+
+    /* eslint-disable dot-notation */
+    LearnerAnswerInfo['createDefaultLearnerAnswerInfo'] = function(answer,
+        answerDetails) {
+    /* eslint-enable dot-notation */
+      return new LearnerAnswerInfo(null, answer, answerDetails,
```

Comment:
Why the default object will have null as id? (is this used anywhere? if not then you can delete it. )

Reply:
Because the id is created in the backend. So when the learner will submit the answer details, its id will be created in the backend.

Reply:
I think this object should always have an id. (Where have you used this method/function?)

Reply:
Actually, this is similar to the way questions are handled in QuestionObjectFactory. Since the ID for the question is only generated in the backend, when it is created, we do set the id to null, but later, when it is fetched, that field is filled.

Reply:
Yeah for creation in the frontend it will be null, but when it will be fetched from the backend there will be a valid id.

Reply:
@anubhavsinha98 please clarify your response. When exactly does the ID for a LearnerAnswerInfo object get created? Please point to the code in question.

Reply:
https://github.com/oppia/oppia/blob/f8d99190f294e8c3d93fc855c5b89b88c583b264/core/domain/stats_services.py#L1165
here the id is being created.

Reply:
Can you please add a comment in the file explaining why the id is null here?